src/client-side-encryption/errors.ts

```diff
@@ -1,63 +1,71 @@
+import { type Document } from '../bson';
+
 /**
- * @class
+ * @public
  * An error indicating that something went wrong specifically with MongoDB Client Encryption
  */
 export class MongoCryptError extends Error {
-  constructor(message, options = {}) {
-    super(message);
-    if (options.cause != null) {
-      this.cause = options.cause;
-    }
+  constructor(message: string, options: { cause?: Error } = {}) {
```

Comment:
```ts
export class MongoCryptError extends Error {
  cause?: Error;
}
```
This was the type of the class in the bindings, so the ctor was already internal, should we keep it the same here? 

Does the inheritance change the type of cause? I think we might want `override cause?: Error` here since if someone has narrowed to this error instance then the cause will be an Error per our ctor argument.

Reply:
Does the inheritance change the type of cause? I think we might want override cause?: Error here since if someone has narrowed to this error instance then the cause will be an Error per our ctor argument.

I'm not sure I follow.  Can you elaborate / provide an example?  `Error.cause` is already an error, no?

Reply:
The cause from TS is defined as `unknown` because it can be any value (just like how `throw 'plain string'` is valid). Our errors will always set an `Error`, so we can make the type more accurate once a variable has been narrowed to one of our error classes.

Reply:
We can address this when we subclass from `MongoError`, because then it'll inherit the narrowed type.